File: daily_python/153_Find_Minimum_in_Rotated_Sorted_Array.py
import os
import sys
from typing import List

class Solution:
    def findMin(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        while l < r:
            if nums[l] < nums[r]:
                return nums[l]
            mid = (l + r) // 2
            if nums[mid] >= nums[l]:
                l = mid + 1
            else:
                r = mid
        return nums[r]

def readDataSet(filename: str):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            dataset.append(nums)
    return dataset

if __name__ == '__main__':
    if (len(sys.argv) == 1):
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for nums in dataset:
        results.append(solution.findMin(nums))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')

File: daily/2026/153_Find_Minimum_in_Rotated_Sorted_Array.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size();
        int left = 0;
        int right = n - 1;
        while(left < right){
            int l = nums[left];
            int r = nums[right];
            if (l <= r) return l;
            int mid = left + (right - left) / 2;
            int m = nums[mid];
            if (m >= l) left = mid + 1;
            else right = mid;
        }
        return nums[right];
    }
};

int main(){
    vector<int> nums = {3,4,5,1,2};
    // vector<int> nums = {4,5,6,7,0,1,2};
    // vector<int> nums = {11,13,15,17};
    // vector<int> nums = {1};
    // vector<int> nums = {2,1};
    Solution S;
    cout << S.findMin(nums) << endl;
    return 0;
}